src/opm/common/utility/FileSystem.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+  Copyright 2019 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/common/utility/FileSystem.hpp>
+
+#include <random>
+
+namespace Opm
+{
+
+std::string unique_path(const std::string& input)
+{
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    auto randchar = [&gen]()
+    {
+        const std::string set = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        std::uniform_int_distribution<> select(0, set.size()-1);
+        return set[select(gen)];
+    };
+
+    std::string ret;
+    for (const char ch : input) {
+        if (ch  == '%')
+            ret += randchar();
+        else
+            ret += ch;
+    }
+
+    return ret;
+}
```

Comment:
Ehhh - if the input string does not contain `%` we will not get any randomness, then the name will not be very unique. Is this expected behavior?

Reply:
i'm not sure what semantics boost::filesystem::unique_path has, i just did whatever was good enough to cover usage.

Reply:
from documentation;
```The unique_path function generates a path name suitable for creating temporary files, including directories. The name is based on a model that uses the percent sign character to specify replacement by a random hexadecimal digit. [Note: The more bits of randomness in the generated path name, the less likelihood of prior existence or being guessed. Each replacement hexadecimal digit in the model adds four bits of randomness. The default model thus provides 64 bits of randomness. This is sufficient for most applications. --end note]```